### PR TITLE
Remove the dependency on the parent report when deciding if report can be approved

### PR DESCRIPTION
**Pull request by mountiny**
### Details

The `IOU.canApproveIOU` method relies on the parent report chat to determine if the report can be approved or not. This is a problem because approvers of expense reports might not necessarily have access to the parent report, or the parent report might not be loaded in local Onyx state, which leads to a bug whereas the approver can't see the approve button.

Luckily we know all the information we need to show the button from the expense report and the policy of the report which the user should have locally.

### Fixed Issues
Related to Slack thread https://expensify.slack.com/archives/C05LX9D6E07/p1724364263668779
N/A

### Tests

@Beamanator can you help me with the test steps here?

1. Create Collect policy (in OldDot), on Advanced Approvals, and make sure `isPolicyExpenseChatEnabled` is `true`
2. Invite members to the policy, like this:
    1. Submitter[User] -> `submitsTo` approver A
    2. Approver A[User] -> `submitsTo` owner
    3. Approver B[User] -> `submitsTo` owner
4. As the submitter, create a report with at least one expense report & submit the report
5. As the owner, change the Submitter's `submitsTo` to Approver B
6. As Approver A, make sure you can see the `Approve` button on the original report from above

- [ ] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by Beamanator** [comment]
Updated to my understanding of what the text steps should be

**Comment by mountiny** [comment]
@ishpaul777 can you check what is the employeeList on the policy and who is the managerID on the report?

**Comment by mountiny** [comment]
Thats definitely odd

**Comment by mountiny** [comment]
@ishpaul777 I dont think these changes will work with pusher updates yet, you might have to sign out and sign back in after each change in the approval chain

**Comment by mountiny** [comment]
So when the report is submitted and awaiting a review from Approver A, that approver should not lose access to the report when the approver is changed.

This is odd, maybe there is something missing in the test steps / policy setup required to replicate the flow we hit with the contributor policy @Beamanator

**Comment by mountiny** [comment]
@ishpaul777 I am not sure still sorry, this is odd. @Beamanator would you be able to help out @ishpaul777 here as you are much more fluent in this flow, I am trying to focus on workspace feeds right now to close the wave-collect

**Comment by mountiny** [comment]
@ishpaul777 I feel like your setup is still not exactly what @JmillsExpensify had in his policy. 

@JmillsExpensify might be able to help us, can you highlight how exactly does your setup look like for the policy/ settings/ approval chain

**Comment by JmillsExpensify** [comment]
Jumping in to try and help with the reproduction steps for this case. I think this should do it, as it's minimally how we discovered this issue: 

 - Admin A create a Control workspace on OldDot. Workspace is policyExpenseChatEnabled
 - Admin invites User B and User C as members.
 - Amin invites User D and User E. User D is the `submitsTo` for B. User E is the `submitsTo` for C
 - Have User B and User C submit reports for approval
 - Confirm that both User D and User E see workspace chats for the relevant members in NewDot, including the ability to approve the report
  - Now in OldDot, have Admin A change the approval workflow. Now User D is the `submitsTo` for C. User E is the `submitsTo` for B
   - Confirm that User D and User E have workspace chats for the right members, and that they are able to approve those respective reports.

**Comment by Beamanator** [comment]
Huh from this video: https://github.com/Expensify/App/pull/47926#issuecomment-2307868416
- I don't reeeeeally understand what's going on - your test user names are so confusing 🤣  and it looks like there's multiple with a name of `sa`? 😅 

Maybe we can pull main, create a new adhoc build, then get @garrettmknight to test this PR tomorrow to see what he gets 🙏

**Comment by Beamanator** [comment]
Ok here's what I'm trying:

- Admin creates a Collect workspace on NewDot, then updates to Control (it updated to non instant submit)
- Admin invites Submitter, Approver 1, Approver 2 as members (via NewDot).
- Admin sets Submitter -> Approver 1 via workflow approvers (in NewDot)
- Have Submitter submit an expense in the workspace chat & submit
    - Verified it submitted to Approver 1 👍 
    - Verified Approver 1 sees workspace chat for Submitter & can access the report & can see Approve button (all in NewDot) 👍 
    - Here's what I saw, first time Approver 1 ever logged in, at this point:
    - Screenshot 2024-08-27 at 7 43 06 PM
- Now in OldDot, have Admin change the approval workflow. Now Approver 2 is the `submitsTo` for Submitter.
    - Verified Approver 2 sees workspace chat for Submitter 👍  & has access to the expense report 👍  & doesn't see Approve button 👍 & Approver 1 is still manager for expense report 👍 
    - Approver 1 SHOULD still have access to the expense report, that is hopefully being fixed in https://github.com/Expensify/Auth/pull/12259 ❌ 
    - Approver 1 SHOULD be able to see the `Approve` button (the fix from this PR) but that can really only be tested once ^ is fixed
    - https://github.com/user-attachments/assets/ce26030f-cdc0-419d-a5c0-ef76404ad471

**Comment by mountiny** [comment]
@Beamanator @garrettmknight running a new build https://github.com/Expensify/App/actions/runs/10592831812

**Comment by garrettmknight** [comment]
Tested and confirmed that Approver A has access to the Approve button even when they've been removed from the workflow + workspace chat.

**Comment by mountiny** [comment]
@ishpaul777 Are you able to complete the checklist and approve the PR? Garrett confirmed that this works for our needs now internally so the pr should be good to go. Thanks!

**Review by ishpaul777 — APPROVED** [review]

**Review by Beamanator — APPROVED** [review]
LGTM

File: src/libs/actions/IOU.ts
Comment (brunovjk) [inline src/libs/actions/IOU.ts:6793]: Previously, we checked the chat report to determine whether it was a policy expense chat, ensuring that such chats were tied to expense reports. Now, we don’t, which seems to have led to this [issue](https://github.com/Expensify/App/issues/52667).

**Pull request merged**
